src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/boutgifter/BoutgifterAndelTilkjentYtelseMapper.kt

```diff
@@ -3,74 +3,32 @@ package no.nav.tilleggsstonader.sak.vedtak.boutgifter
 import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.AndelTilkjentYtelse
 import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.Satstype
 import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.StatusIverksetting
-import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.TypeAndel
 import no.nav.tilleggsstonader.sak.util.datoEllerNesteMandagHvisLørdagEllerSøndag
 import no.nav.tilleggsstonader.sak.vedtak.boutgifter.domain.BeregningsresultatBoutgifter
-import no.nav.tilleggsstonader.sak.vedtak.boutgifter.domain.BeregningsresultatForLøpendeMåned
-import java.time.LocalDate
-import kotlin.collections.component1
-import kotlin.collections.component2
 
 object BoutgifterAndelTilkjentYtelseMapper {
     fun finnAndelTilkjentYtelse(
         saksbehandling: Saksbehandling,
         beregningsresultat: BeregningsresultatBoutgifter,
-    ): List<AndelTilkjentYtelse> {
-        val andeler =
-            beregningsresultat.perioder
-                .groupBy { it.grunnlag.utbetalingsdato }
-                .entries
-                .sortedBy { (utbetalingsdato, _) -> utbetalingsdato }
-                .flatMap { (utbetalingsdato, perioder) ->
-                    val førstePerioden = perioder.first()
-                    val satsBekreftet = førstePerioden.grunnlag.makssatsBekreftet
-
-                    feilHvisIkke(perioder.all { it.grunnlag.makssatsBekreftet == satsBekreftet }) {
-                        "Alle perioder for et utbetalingsdato må være bekreftet eller ikke bekreftet"
-                    }
-
-                    mapTilAndeler(
-                        perioder = perioder,
-                        saksbehandling = saksbehandling,
-                        utbetalingsdato = utbetalingsdato.datoEllerNesteMandagHvisLørdagEllerSøndag(),
-                        satsBekreftet = satsBekreftet,
-                    )
-                }
-        return andeler
-    }
-
-    /**
-     * Andeler grupperes per [TypeAndel], sånn at hvis man har 2 ulike målgrupper men som er av samme [TypeAndel]
-     * så summeres beløpet sammen for disse 2 andelene
-     * Hvis man har 2 [BeregningsresultatForLøpendeMåned] med med 2 ulike [TypeAndel]
-     * så blir det mappet til ulike andeler for at regnskapet i økonomi skal få riktig type for gitt utbetalingsmåned
-     */
-    private fun mapTilAndeler(
-        perioder: List<BeregningsresultatForLøpendeMåned>,
-        saksbehandling: Saksbehandling,
-        utbetalingsdato: LocalDate,
-        satsBekreftet: Boolean,
-    ): List<AndelTilkjentYtelse> {
-        val førsteUkedagIMåneden = utbetalingsdato.tilFørsteDagIMåneden().datoEllerNesteMandagHvisLørdagEllerSøndag()
-        return perioder
-            .groupBy { it.grunnlag.målgruppe.tilTypeAndel(Stønadstype.BOUTGIFTER) }
-            .map { (typeAndel, perioder) ->
+    ): List<AndelTilkjentYtelse> =
+        beregningsresultat.perioder
+            .sortedBy { it.fom }
```

Comment:
Då periode er av type `BeregningsresultatForLøpendeMåned` som er en `Periode<LocalDate>` som er en `Comparable<LocalDate>` så er det tilstrekkelig å kun ha `.sorted()`
```suggestion
            .sorted()
```